packages/api/cms-api/src/mail-templates/mail-template.service.ts

```diff
@@ -0,0 +1,98 @@
+import { DiscoveryService } from "@golevelup/nestjs-discovery";
+import { EntityManager } from "@mikro-orm/postgresql";
+import { Injectable } from "@nestjs/common";
+import { Options as MailOptions } from "nodemailer/lib/mailer";
+import { createIntl, createIntlCache, IntlCache, IntlShape } from "react-intl";
+import { ContentScope } from "src/user-permissions/interfaces/content-scope.interface";
+
+import { MailerService } from "../mailer/mailer.service";
+import { MAIL_TEMPLATE_METADATA_KEY, MailTemplateMetadata } from "./mail-template.decorator";
+
+export type MailTemplateInterface<T> = {
+    id: string;
+    type: string;
+    name: string;
+
+    availableForScopes?: () => Promise<ContentScope[]>;
```

Comment:
Topic: mail style testing in admin: probably needs some scope checks

Reply:
for testing but also for scope-specific templates. think about a tool tracking orders of vehicles of different brands, and the "superuser" checking if the mail displays the correct values.

but this is one of the aspects where this module lacks experience, because this is currently not fully tested and probably still missing a lot.

Reply:
As discussed, please remove the scope-specific parts for now.